sundials@7.0.0 +rocm w/ ROCm 6.1.1: sundials_hip_kernels.hip.hpp:357:27: error: reference to __host__ variable 'MAX_WARPS' in __device__ function

### Steps to reproduce the issue

`sundials@7.7.0 +rocm` build fails using:
* spack@develop (641075539c0a888d4378bd7dfdb19a872c5c42de from `Mon Jun 3 14:37:51 2024 -0700`)
* ROCm 6.1.1 installed via apt, externally defined via `packages:` config
* Ubuntu 22.04, X86_64

Spack environment: [spack.yaml.txt](https://github.com/user-attachments/files/15687547/spack.yaml.txt)

<details><summary>Concretization</summary>

<pre>
 -   4mdq6al  sundials@7.0.0%gcc@11.4.0+ARKODE+CVODE+CVODES+IDA+IDAS+KINSOL~cuda+examples+examples-install~f2003~fcmix~ginkgo~hypre~int64~ipo~klu~kokkos~kokkos-kernels~lapack~magma~monitoring+mpi~openmp~petsc~profiling~pthread~raja+rocm+shared+static~superlu-dist~superlu-mt~sycl~trilinos amdgpu_target=gfx90a build_system=cmake build_type=Release cstd=99 cxxstd=14 generator=make logging-level=2 precision=double arch=linux-ubuntu22.04-x86_64_v3
[+]  45jczu5      ^cmake@3.29.2%gcc@11.4.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-ubuntu22.04-x86_64_v3
[+]  7wjyjza          ^curl@8.7.1%gcc@11.4.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-ubuntu22.04-x86_64_v3
[+]  zc6oo3e              ^nghttp2@1.62.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  sycoqix                  ^diffutils@3.10%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  r3wk5bb              ^openssl@3.3.0%gcc@11.4.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu22.04-x86_64_v3
[+]  t2brqss                  ^ca-certificates-mozilla@2023-05-30%gcc@11.4.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]  vqvzn64              ^perl@5.38.2%gcc@11.4.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-ubuntu22.04-x86_64_v3
[+]  sw7gp6x                  ^berkeley-db@18.1.40%gcc@11.4.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu22.04-x86_64_v3
[+]  tmq3cku                  ^bzip2@1.0.8%gcc@11.4.0~debug~pic+shared build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]  xucr42b                  ^gdbm@1.23%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  sqasgz3          ^ncurses@6.5%gcc@11.4.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-ubuntu22.04-x86_64_v3
[+]  54554gr          ^zlib-ng@2.1.6%gcc@11.4.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  hshzy76      ^gcc-runtime@11.4.0%gcc@11.4.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[e]  a7drdl4      ^glibc@2.35%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  aevr47s      ^gmake@4.4.1%gcc@11.4.0~guile build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[e]  abm43bu      ^hip@6.1.1%gcc@11.4.0~cuda+rocm build_system=cmake build_type=Release generator=make patches=1f65dfe,6bbe83b arch=linux-ubuntu22.04-x86_64_v3
[e]  qyqwwse      ^hsa-rocr-dev@6.1.1%gcc@11.4.0~asan+image+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[e]  l4maeaq      ^llvm-amdgpu@6.1.1%gcc@11.4.0~link_llvm_dylib~llvm_dylib+rocm-device-libs build_system=cmake build_type=Release generator=ninja patches=b4774ca arch=linux-ubuntu22.04-x86_64_v3
[+]  q73zij2      ^mpich@4.2.1%gcc@11.4.0~argobots~cuda+fortran~hwloc+hydra+libxml2+pci~rocm+romio~slurm~vci~verbs~wrapperrpath~xpmem build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=default arch=linux-ubuntu22.04-x86_64_v3
[+]  43jtbdg          ^findutils@4.9.0%gcc@11.4.0 build_system=autotools patches=440b954 arch=linux-ubuntu22.04-x86_64_v3
[+]  grokknt          ^libfabric@1.21.0%gcc@11.4.0~debug~kdreg~uring build_system=autotools fabrics=sockets,tcp,udp arch=linux-ubuntu22.04-x86_64_v3
[+]  cinntmr          ^libpciaccess@0.17%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  ekglkxs              ^libtool@2.4.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  eu34nvi              ^util-macros@1.20.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  blvm6hb          ^libxml2@2.10.3%gcc@11.4.0+pic~python+shared build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  imwcbq7              ^xz@5.4.6%gcc@11.4.0~pic build_system=autotools libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[+]  mv2wke5          ^pkgconf@2.2.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  x2wkmjd          ^yaksa@0.3%gcc@11.4.0~cuda~rocm build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  tfy5aww              ^autoconf@2.72%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  c5h3rwv              ^automake@1.16.5%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  iloyaza              ^m4@1.4.19%gcc@11.4.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu22.04-x86_64_v3
[+]  jv4abt3                  ^libsigsegv@2.14%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  rmokzl2              ^python@3.11.9%gcc@11.4.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-ubuntu22.04-x86_64_v3
[+]  6sx3gbp                  ^expat@2.6.2%gcc@11.4.0+libbsd build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  wfvladn                      ^libbsd@0.12.2%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  i3snxv7                          ^libmd@1.0.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  3ea5rwn                  ^gettext@0.22.5%gcc@11.4.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  a4yvhcg                      ^tar@1.34%gcc@11.4.0 build_system=autotools zip=pigz arch=linux-ubuntu22.04-x86_64_v3
[+]  5qg7ajj                          ^pigz@2.8%gcc@11.4.0 build_system=makefile arch=linux-ubuntu22.04-x86_64_v3
[+]  fz3fbgw                          ^zstd@1.5.6%gcc@11.4.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[+]  ounz353                  ^libffi@3.4.6%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  wq6feyo                  ^libxcrypt@4.4.35%gcc@11.4.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-ubuntu22.04-x86_64_v3
[+]  bo6z4bc                  ^readline@8.2%gcc@11.4.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu22.04-x86_64_v3
[+]  jvljkqx                  ^sqlite@3.43.2%gcc@11.4.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  vl2hqwb                  ^util-linux-uuid@2.40.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
</pre>
</details>

Error:
```
$> spack -e . install
...
==> Installing sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp [19/19]
==> No binary for sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp found: installing from source
==> Using cached archive: /spack/var/spack/cache/_source-cache/archive/d7/d762a7950ef4097fbe9d289f67a8fb717a0b9f90f87ed82170eb5c36c0a07989.tar.gz
==> No patches needed for sundials
==> sundials: Executing phase: 'cmake'
==> sundials: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j32'

43 errors found in build log:
     2661    [ 62%] Building C object examples/idas/parallel/CMakeFiles/idasHeat2D_kry_p.dir/idasHeat2D_kry_p.c.o
     2662    cd /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al/examples/idas/parallel && /spack/opt/spack/linux-ubuntu22.04-x
             86_64_v3/gcc-11.4.0/mpich-4.2.1-q73zij2acabvp27dqu6eakamcxfxcsja/bin/mpicc  -I/tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-s
             rc/include -I/tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al/include -I/tmp/root/spack-stage/spack-stage-sundials
             -7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-src/src/sundials -I/tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al/
             src/sundials -isystem /spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/mpich-4.2.1-q73zij2acabvp27dqu6eakamcxfxcsja/include -O3 -DNDEBUG -std=gnu99 -MD -MT exam
             ples/idas/parallel/CMakeFiles/idasHeat2D_kry_p.dir/idasHeat2D_kry_p.c.o -MF CMakeFiles/idasHeat2D_kry_p.dir/idasHeat2D_kry_p.c.o.d -o CMakeFiles/idasHeat2D_kry_p.dir/i
             dasHeat2D_kry_p.c.o -c /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-src/examples/idas/parallel/idasHeat2D_kry_p.c
     2663    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al'
     2664    [ 62%] Built target cvDiurnal_kry_bbd_p
     2665    In file included from /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-src/src/nvector/hip/nvector_hip.hip.cpp:23:
     2666    In file included from /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-src/include/nvector/nvector_hip.h:24:
  >> 2667    /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-src/include/sundials/sundials_hip_policies.hpp:36:64: error: use of undeclared
             identifier 'WARP_SIZE'
     2668       36 | constexpr const sunindextype MAX_WARPS      = MAX_BLOCK_SIZE / WARP_SIZE;
     2669          |                                                                ^
     2670    /opt/rocm-6.1.1/llvm/bin/clang -O3 -DNDEBUG -Wl,-rpath -Wl,/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/mpich-4.2.1-q73zij2acabvp27dqu6eakamcxfxcsja/lib CMa
             keFiles/idasRoberts_ASAi_dns.dir/idasRoberts_ASAi_dns.c.o -o idasRoberts_ASAi_dns  -Wl,-rpath,/tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbi
             ps2wjihp/spack-build-4mdq6al/src/idas:/tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al/src/nvector/serial:/tmp/roo
             t/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al/src/sundials:/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/mpic
             h-4.2.1-q73zij2acabvp27dqu6eakamcxfxcsja/lib ../../../src/idas/libsundials_idas.so.6.0.0 ../../../src/nvector/serial/libsundials_nvecserial.so.7.0.0 -lm ../../../src/s
             undials/libsundials_core.so.7.0.0 /spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/mpich-4.2.1-q73zij2acabvp27dqu6eakamcxfxcsja/lib/libmpi.so
     2671    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al'
     2672    make  -f examples/kinsol/serial/CMakeFiles/kinFerTron_dns.dir/build.make examples/kinsol/serial/CMakeFiles/kinFerTron_dns.dir/depend
     2673    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al'
     2674    [ 63%] Linking C executable idaFoodWeb_kry_bbd_p
     2675    make[2]: Entering directory '/tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al'
     2676    cd /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al && /spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/cma
             ke-3.29.2-45jczu57cadhtdehiz423qzpr73zsitr/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjih
             p/spack-src /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-src/examples/kinsol/serial /tmp/root/spack-stage/spack-stage-sundia
             ls-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al/exa
             mples/kinsol/serial /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al/examples/kinsol/serial/CMakeFiles/kinFerTron_
             dns.dir/DependInfo.cmake "--color="
     2677    [ 63%] Building C object examples/kinsol/serial/CMakeFiles/kinAnalytic_fp.dir/kinAnalytic_fp.c.o
     2678    cd /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al/examples/ida/parallel && /spack/opt/spack/linux-ubuntu22.04-x8
             6_64_v3/gcc-11.4.0/cmake-3.29.2-45jczu57cadhtdehiz423qzpr73zsitr/bin/cmake -E cmake_link_script CMakeFiles/idaFoodWeb_kry_bbd_p.dir/link.txt --verbose=1
     2679    cd /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al/examples/kinsol/serial && /opt/rocm-6.1.1/llvm/bin/clang  -I/t
             mp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-src/include -I/tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasja
             wm7tbips2wjihp/spack-build-4mdq6al/include -I/tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-src/src/sundials -I/tmp/root/spack
             -stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-build-4mdq6al/src/sundials -isystem /spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/mp
             ich-4.2.1-q73zij2acabvp27dqu6eakamcxfxcsja/include -O3 -DNDEBUG -std=gnu99 -MD -MT examples/kinsol/serial/CMakeFiles/kinAnalytic_fp.dir/kinAnalytic_fp.c.o -MF CMakeFil
             es/kinAnalytic_fp.dir/kinAnalytic_fp.c.o.d -o CMakeFiles/kinAnalytic_fp.dir/kinAnalytic_fp.c.o -c /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm
             7tbips2wjihp/spack-src/examples/kinsol/serial/kinAnalytic_fp.c
  >> 2680    /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-src/include/sundials/sundials_hip_policies.hpp:157:36: error: use of undeclared
              identifier 'WARP_SIZE'
     2681      157 |     if (blockDim < 1 || blockDim % WARP_SIZE)
     2682          |                                    ^
     2683    [ 63%] Built target cvDiurnal_kry_mpimanyvec
     2684    make  -f examples/kinsol/serial/CMakeFiles/kinFoodWeb_kry.dir/build.make examples/kinsol/serial/CMakeFiles/kinFoodWeb_kry.dir/depend
  >> 2685    /tmp/root/spack-stage/spack-stage-sundials-7.0.0-4mdq6alubj65sxasjawm7tbips2wjihp/spack-src/include/sundials/sundials_hip_policies.hpp:201:36: error: use of undeclared
              identifier 'WARP_SIZE'
     2686      201 |     if (blockDim < 1 || blockDim % WARP_SIZE)
     2687          |
```

### Error message

<details><summary>Error message</summary>

<pre> 
... see above
</pre></details>
 

### Information on your system

* **Spack:** 0.23.0.dev0 (136308c8fdc3d3d0593d563103f34e60ab15a5d6)
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/user-attachments/files/15687543/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/15687544/spack-build-out.txt)

@balos1 @cswoodward @gardner48

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/sundials/package.py
+++ b/var/spack/repos/builtin/packages/sundials/package.py
@@ -284,6 +284,10 @@ class Sundials(CMakePackage, CudaPackage, ROCmPackage):
     # ==========================================================================
     # Patches
     # ==========================================================================
+    # https://github.com/LLNL/sundials/pull/434
+    # https://github.com/LLNL/sundials/pull/437
+    patch("sundials-hip-platform.patch", when="@7.0.0 +rocm")
+
     # https://github.com/spack/spack/issues/29526
     patch("nvector-pic.patch", when="@6.1.0:6.2.0 +rocm")
 

--- a/var/spack/repos/builtin/packages/sundials/sundials-hip-platform.patch
+++ b/var/spack/repos/builtin/packages/sundials/sundials-hip-platform.patch
@@ -1,0 +1,33 @@
+diff -ruN spack-src/cmake/SundialsSetupHIP.cmake spack-src-patched/cmake/SundialsSetupHIP.cmake
+--- spack-src/cmake/SundialsSetupHIP.cmake	2024-02-29 22:47:30.000000000 +0000
++++ spack-src-patched/cmake/SundialsSetupHIP.cmake	2024-06-07 15:22:57.920619263 +0000
+@@ -32,9 +32,9 @@
+ 
+ if(NOT DEFINED HIP_PLATFORM)
+   if(NOT DEFINED ENV{HIP_PLATFORM})
+-    set(HIP_PLATFORM "hcc" CACHE STRING "HIP platform (hcc, nvcc)")
++    set(HIP_PLATFORM "amd" CACHE STRING "HIP platform (amd, nvidia)")
+   else()
+-    set(HIP_PLATFORM "$ENV{HIP_PLATFORM}" CACHE STRING "HIP platform (hcc, nvcc)")
++    set(HIP_PLATFORM "$ENV{HIP_PLATFORM}" CACHE STRING "HIP platform (amd, nvidia)")
+   endif()
+ endif()
+ 
+diff -ruN spack-src/include/sundials/sundials_hip_policies.hpp spack-src-patched/include/sundials/sundials_hip_policies.hpp
+--- spack-src/include/sundials/sundials_hip_policies.hpp	2024-02-29 22:47:30.000000000 +0000
++++ spack-src-patched/include/sundials/sundials_hip_policies.hpp	2024-06-07 15:23:15.752724851 +0000
+@@ -27,10 +27,12 @@
+ namespace sundials {
+ namespace hip {
+ 
+-#if defined(__HIP_PLATFORM_HCC__)
++#if defined(__HIP_PLATFORM_HCC__) || defined(__HIP_PLATFORM_AMD__)
+ constexpr const sunindextype WARP_SIZE = 64;
+-#elif defined(__HIP_PLATFORM_NVCC__)
++#elif defined(__HIP_PLATFORM_NVCC__) || defined(__HIP_PLATFORM_NVDIA__)
+ constexpr const sunindextype WARP_SIZE = 32;
++#else
++#error "Unknown HIP_PLATFORM, report to github.com/LLNL/sundials/issues"
+ #endif
+ constexpr const sunindextype MAX_BLOCK_SIZE = 1024;
+ constexpr const sunindextype MAX_WARPS      = MAX_BLOCK_SIZE / WARP_SIZE;

